### PR TITLE
added support for amazon linux and renamed tests.yml

### PR DESCRIPTION
**Pull request by redraincatching**
superseding changes made in pr https://github.com/coturn/coturn/pull/1394 on request of @jonesmz

dependent on pr https://github.com/coturn/coturn/pull/1399 - uses the composite action for dependency installation i wrote there

added YAML file for amazon linux and renames tests.yml to ubuntu_tests.yml for clarity

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -2,10 +2,10 @@ name: "CodeQL"
 
 on:
   push:
-    branches: [ 'master' ]
+    # branches: [ 'master' ]
```
Comment (ggarber) [inline .github/workflows/codeql.yml:5]: Is this change related to this PR?

**Review by ggarber — APPROVED** [review]

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (redraincatching) [reply .github/workflows/codeql.yml]: i don't believe it was, have resolved it now

**Comment by eakraly** [comment]
Hi @redraincatching ! Could you please rebase your changes so we can merge them?

**Comment by redraincatching** [comment]
@eakraly rebased and ready to merge !

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -2,6 +2,7 @@ name: "CodeQL"
 
 on:
   push:
+    branches: [ 'master' ]
```
Comment (jonesmz) [inline .github/workflows/codeql.yml:5]: IMHO don't add this, i think this is a rebase / merge artifact.

Comment (redraincatching) [reply .github/workflows/codeql.yml]: it is, got confused because of the first reply to this pr

File: .github/workflows/ubuntu_tests.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ 'ubuntu:16.04', 'ubuntu:18.04', 'ubuntu:20.04', 'ubuntu:22.04' ]
+        os: [ 'ubuntu:18.04', 'ubuntu:20.04', 'ubuntu:22.04' ]
```
Comment (eakraly) [inline .github/workflows/ubuntu_tests.yml:13]: Please return ubuntu 16.04 - we are still not ready to let it go

File: .github/workflows/amazon_linux_tests.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    runs-on: ubuntu-latest
+    container: ${{ matrix.os }}
+
+    steps:
+    - name: Install Dependencies
+      run: |
+        yum install -y gcc make gzip tar openssl-devel libevent-devel
+    # Delay checkout until after dependencies have been installed. Amazon linux is very minimal and lacks basic stuff by default.
+    # use v3 of checkout until the weird container, nodejs, glibc issue is fixed
+    - uses: actions/checkout@v3
```
Comment (jonesmz) [inline .github/workflows/amazon_linux_tests.yml:25]: version bump to v4

Comment (redraincatching) [reply .github/workflows/amazon_linux_tests.yml]: the glibc errors still seem to be present for amazon linux 2 for version 4

File: .github/workflows/ubuntu_tests.yml (hunk 3 above)
Comment (redraincatching) [reply .github/workflows/ubuntu_tests.yml]: that has been returned
